### DISCUSSION TITLE
2017 Google Pixelbook (intel i7, intel iris 615?)

### DISCUSSION THREAD

**Discussion by RogerDass** [leejet/stable-diffusion.cpp#121] [Benchmark]
Hello!

Thanks for making this amazing project!

So i'm running this with realistic vision v1.5 checkpoint. I'm getting ~75 s/it with OpenBLAS enabled.
Any idea how to speed that up significantly?

With ComfyUI on the same machine in CPU mode, I'm getting ~30 s/it and it takes 10 min to generate 512x512 image with 20 steps. 

What's causing such a large performance difference?

Do you know if there's a way to get some basic OpenGL 3 acceleration for some of the tensor ops?

**Comment by Cyberhan123** [comment]
My observation is that `ggml` does not achieve maximum convolution optimization, my solution is to use turbo model or configure quantization, such as q8_0

**Comment by Amin456789** [comment]
use lcm lora sdv1.5 with 4 steps and taesd  it be much faster

**Comment by Green-Sky** [comment]
> ~75 it/s 

pretty sure you meant s/it

**Comment by RogerDass** [comment]
Yeah s/it...

Please keep in mind that ComfyUI is using f32 and not any lower quantization with 20 steps and is more than 2X faster.

Anyone know what tensorvision-cpu is doing that's so much faster than gglm?

**Comment by FSSRepo** [comment]
@RogerDass It's just that PyTorch implements more optimized convolution algorithms that are too complex to implement in ggml. That's why PyTorch is quite heavy; instead of reinventing the wheel, they reuse existing code to avoid unnecessary complications.
